mnefun/_simulate.py

```diff
@@ -121,15 +122,20 @@ def _make_forward_solutions(info, mri, src, bem, bem_eog, dev_head_ts, mindist,
 
     # make a new dict with the relevant information
     mri_id = dict(machid=np.zeros(2, np.int32), version=0, secs=0, usecs=0)
-    info = dict(nchan=info['nchan'], chs=info['chs'], comps=info['comps'],
+    info = Info(nchan=info['nchan'], chs=info['chs'], comps=info['comps'],
                 ch_names=info['ch_names'],
                 mri_file='', mri_id=mri_id, meas_file='',
                 meas_id=None, working_dir=os.getcwd(),
                 command_line='', bads=info['bads'])
 
     # Only get the EEG channels here b/c we can do MEG later
-    _, _, eegels, _, eegnames, _ = \
-        _prep_channels(info, False, True, True, verbose=False)
+    if len(pick_types(info, meg=False, eeg=True)) > 0:
+        do_eeg = True
+        _, _, eegels, _, eegnames, _ = \
+            _prep_channels(info, False, True, True, verbose=False)
```

Comment:
My PR in progress (#2355) will immediately break this once it get's merged since I split up `_prep_channels` :/